pennylane/operation.py

```diff
@@ -1061,9 +1061,14 @@ def decomposition(self):
         Returns:
             list[Operator]: decomposition of the operator
         """
-        return self.compute_decomposition(
+        decomp = self.compute_decomposition(
```

Comment:
I think I ultimately intend on reverting _all_ changes to this file before merging, but while testing it has been useful to have this expansion, and raise this error. I worry that this validation in `Operator.decomposition` will be too much of a performance impact to keep.